### PR TITLE
feat(connectors): add OAuth connector resource

### LINKED ISSUE #184: Add OAuth connectors as a CLI resource
## Overview

OAuth connectors allow Base44 apps to integrate with external services (Google Calendar, Slack, LinkedIn, etc.) via OAuth. This document describes the proposed CLI resource implementation.

---

### Backend (apper)

- [ ] **Task 0**: Implement `/sync` endpoint — base44-dev/apper#3374

### CLI

- [x] **Task 1**: Resource file parsing (Zod schemas, file reader) — #189
- [x] **Task 2**: API client methods (`list`, `status`, `remove`, `sync`) -- #190 
- [x] **Task 3**: Push comparison logic -- #191 
- [x] **Task 4**: OAuth flow handling (URL display, polling, timeout) -- #192 
- [x] **Task 5**: Push command integration + summary output -- #193

### Task Details

| Task | Description | Blocked By |
|------|-------------|------------|
| Task 0 | Backend `/sync` endpoint with exact scope matching | — |
| Task 1 | Zod schemas for connector resources, JSONC file reader | — |
| Task 2 | API client: `syncConnector()`, `listConnectors()`, `getOAuthStatus()`, `removeConnector()` | Task 0 (for sync) |
| Task 3 | Compare local vs upstream, determine actions (sync/delete) | Task 0 |
| Task 4 | Display auth URL, poll for completion, handle edge cases | Task 2 |
| Task 5 | Wire into push command, colored summary output | Tasks 3, 4 |

---

### Key Concepts

| Term | Description |
|------|-------------|
| **AppIntegration** | MongoDB model storing connector state per app |
| **Composio** | Third-party service handling OAuth flows, token storage, and refresh |
| **Connected Account** | A user's authorized connection to an external service |

### Supported Integration Types

```
googlecalendar, googledrive, gmail, googlesheets, googledocs, googleslides,
slack, notion, salesforce, hubspot, linkedin, tiktok
```

### Connector States

| Status | Description |
|--------|-------------|
| `ACTIVE` | Connected and functional |
| `DISCONNECTED` | Soft-deleted, preserves scopes for reconnection |
| `EXPIRED` | Token expired, needs re-auth |

---

### File Structure

```
connectors/
├── googlecalendar.jsonc
├── slack.jsonc
└── notion.jsonc
```

### Resource Schema

```jsonc
// connectors/googlecalendar.jsonc
{
  "type": "googlecalendar",
  "scopes": [
    "https://www.googleapis.com/auth/calendar.readonly",
    "https://www.googleapis.com/auth/calendar.events"
  ]
}
```

- One file per connector, named by integration type
- **Stateless** - contains only type and scopes, no status field
- Uses JSONC (JSON with comments) for consistency with other CLI resources
- **No pull support** - connectors are push-only

---

### The `/sync` Endpoint

The CLI uses a dedicated `/sync` endpoint designed for declarative state management. Unlike the `/initiate` endpoint (used by Apper's AI for incremental scope expansion), `/sync` provides:

1. **Exact scope matching** - returns `already_authorized=true` only when scopes match exactly
2. **Scope replacement** - never merges scopes, always replaces with requested scopes
3. **Built-in scope enhancement** - backend handles auto-added scopes internally

### Comparison Matrix

| Local File | Upstream State | Scopes Match? | Action |
|------------|----------------|---------------|--------|
| Exists | Not exists | N/A | Call `/sync` → auth URL + poll |
| Exists | `DISCONNECTED` | N/A | Call `/sync` → auth URL + poll |
| Exists | `EXPIRED` | N/A | Call `/sync` → auth URL + poll |
| Exists | `ACTIVE` | Yes | Call `/sync` → `already_authorized=true` (no-op) |
| Exists | `ACTIVE` | No | Call `/sync` → auth URL + poll (scopes replaced) |
| Not exists | Exists (any) | N/A | Hard delete upstream |

### How `/sync` Works

1. CLI sends `{ integration_type, scopes }` to `/sync`
2. Backend enhances scopes internally (adds identity scopes like `email`)
3. Backend compares `enhanced(requested)` with existing approved scopes (exact equality)
4. If exact match → returns `{ already_authorized: true }`
5. If different → replaces auth config scopes, creates new connection, returns `{ redirect_url, connection_id }`

This is simpler than the AI flow because:
- No client-side scope enhancement needed
- No delete-then-create dance
- Single atomic operation

### Post-Auth Verification

After OAuth completes, verify the approved scopes match what was expected:

1. Fetch connector from `/list` endpoint
2. Compare `approved_scopes` with `requested_scopes` from JSONC
3. If not equal → show `SCOPE_MISMATCH` (user gave partial consent)

Note: The `/list` endpoint returns the **enhanced** scopes (including auto-added). The CLI should compare against the full approved set, not just the user-specified scopes.

**Possible outcomes:**

| Approved vs Expected | Result |
|---------------------|--------|
| Scopes match | ✓ ACTIVE |
| User deselected scopes during consent | ⚠ SCOPE_MISMATCH |

### Auth Flow Behavior

- **Sequential auth**: If multiple connectors need auth, prompt one at a time (interactive state machine)
- **Polling timeout**: Same as device login auth timeout

---

## Push Summary Output

Uses `@clack/prompts` logging with colors (consistent with existing CLI patterns):

```
Connectors push summary:
  - googlecalendar: active (3 scopes)
  - slack: active (4 scopes, re-authed)
  - linkedin: scope mismatch (requested 5, approved 3)
  - notion: auth not completed
  - hubspot: deleted (no local definition)

Some connectors need attention:
  - linkedin: Approved scopes differ from requested. Update connectors/linkedin.jsonc or run push again.
  - notion: Authentication not completed. Run push to retry.
```

Status text will be colored via chalk:
- **active** → green
- **scope mismatch** → yellow
- **auth not completed / failed** → red
- **deleted** → dim/gray

### Summary Status States

| Status | Color | Description |
|--------|-------|-------------|
| `ACTIVE` | green | Connector active, scopes match |
| `SCOPE_MISMATCH` | yellow | Active but approved scopes ≠ requested (partial consent) |
| `PENDING_AUTH` | red | Auth URL shown but user didn't complete |
| `AUTH_FAILED` | red | OAuth flow failed |
| `DELETED` | dim | Removed from upstream (no local file) |
| `DIFFERENT_USER` | red | Another user already authorized this connector |

---

## API Endpoints

| Endpoint | Method | Description |
|----------|--------|-------------|
| `/api/apps/{app_id}/external-auth/sync` | POST | **CLI endpoint** - sync connector with exact scopes |
| `/api/apps/{app_id}/external-auth/list` | GET | List all connectors |
| `/api/apps/{app_id}/external-auth/status` | GET | Poll for OAuth completion |
| `/api/apps/{app_id}/external-auth/integrations/{type}/remove` | DELETE | Hard delete |

### Sync Endpoint (CLI-specific)

```
POST /api/apps/{app_id}/external-auth/sync
{
  "integration_type": "googlecalendar",
  "scopes": ["https://www.googleapis.com/auth/calendar.readonly"]
}

// Response - needs auth
{
  "redirect_url": "https://accounts.google.com/o/oauth2/...",
  "connection_id": "abc123",
  "already_authorized": false
}

// Response - already authorized with exact scopes
{
  "redirect_url": null,
  "connection_id": null,
  "already_authorized": true
}

// Response - different user error
{
  "redirect_url": null,
  "connection_id": null,
  "already_authorized": false,
  "error": "different_user",
  "error_message": "Integration already authorized by another user",
  "other_user_email": "other@example.com"
}
```

**Key differences from `/initiate`:**
- Exact scope comparison (not subset)
- Replaces scopes (not merges)
- Handles scope enhancement server-side

### Polling

```
GET /status?integration_type=googlecalendar&connection_id=abc123
→ {"status": "ACTIVE" | "FAILED" | "PENDING"}
```

Poll until `ACTIVE` or `FAILED`. Suggested interval: 2 seconds.

---

## Edge Cases

1. **Partial Consent**: OAuth provider may approve fewer scopes than requested → show `SCOPE_MISMATCH`

2. **Different User**: Only one user can auth per connector per app. If another user already authorized, show their email and suggest they disconnect first.

3. **Scope Validation**: Scopes are NOT validated by apper - the OAuth provider validates during consent. Invalid scopes cause errors on the provider's consent screen.

4. **Provider doesn't return scopes**: Some OAuth providers (e.g., Facebook) don't return approved scopes in token response. Apper falls back to requested (enhanced) scopes.

---

## Auto-Added Scopes (Reference)

These scopes are added automatically by the backend for identity extraction. Users don't need to include them in JSONC files - the `/sync` endpoint handles enhancement internally.

| Integration | Auto-added Scopes |
|-------------|-------------------|
| All Google (Calendar, Drive, Gmail, Sheets, Docs, Slides) | `email` |
| Slack | `users:read`, `users:read.email` |
| Salesforce | `openid`, `profile`, `email` |
| HubSpot | `oauth` |
| LinkedIn | `openid`, `profile`, `email` |
| TikTok | `user.info.basic` |
| Notion | *(none)* |

---

### Google Calendar
| Scope | Description |
|-------|-------------|
| `https://www.googleapis.com/auth/calendar.readonly` | Read-only access |
| `https://www.googleapis.com/auth/calendar.events` | Read/write events |
| `https://www.googleapis.com/auth/calendar` | Full access |

### Google Drive
| Scope | Description |
|-------|-------------|
| `https://www.googleapis.com/auth/drive.file` | App-created/selected files only (**only supported**) |
| `https://www.googleapis.com/auth/drive.readonly` | Read all files |

### Gmail
| Scope | Description |
|-------|-------------|
| `https://www.googleapis.com/auth/gmail.send` | Send emails |
| `https://www.googleapis.com/auth/gmail.readonly` | Read emails |
| `https://www.googleapis.com/auth/gmail.modify` | Read/write/send |

### Google Sheets
| Scope | Description |
|-------|-------------|
| `https://www.googleapis.com/auth/spreadsheets` | Read/write |
| `https://www.googleapis.com/auth/spreadsheets.readonly` | Read-only |

### Google Docs
| Scope | Description |
|-------|-------------|
| `https://www.googleapis.com/auth/documents` | Read/write |
| `https://www.googleapis.com/auth/documents.readonly` | Read-only |

### Google Slides
| Scope | Description |
|-------|-------------|
| `https://www.googleapis.com/auth/presentations` | Read/write |
| `https://www.googleapis.com/auth/presentations.readonly` | Read-only |

### Slack
| Scope | Description |
|-------|-------------|
| `chat:write` | Send messages |
| `channels:read` | Read channel info |
| `channels:history`, `groups:read`, `im:read`, `mpim:read` | Read various channels |
| `files:read`, `files:write` | File access |

*Note: User token integration only. Do not request bot scopes.*

### Notion
Notion uses a single-scope model - users select pages/databases in the OAuth UI. No scopes needed in JSONC.

### Salesforce
| Scope | Description |
|-------|-------------|
| `api` | API access |
| `refresh_token` | Offline access |

### HubSpot
| Scope | Description |
|-------|-------------|
| `crm.objects.contacts.read/write` | Contacts access |
| `crm.objects.deals.read/write` | Deals access |
| `crm.objects.companies.read/write` | Companies access |

### LinkedIn
| Scope | Description |
|-------|-------------|
| `r_basicprofile`, `r_profile_basicinfo` | Profile read |
| `w_member_social` | Post as member |
| `w_organization_social`, `r_organization_social` | Org social |
| `r_organization_admin`, `rw_organization_admin` | Org admin |
| `r_ads`, `r_ads_reporting`, `rw_ads` | Ads access |
| `r_verify`, `r_1st_connections_size` | Verification, connections |

### TikTok
| Scope | Description |
|-------|-------------|
| `user.info.profile` | Links, bio, verification |
| `user.info.stats` | Follower/video counts |
| `video.list` | Read public videos |
| `artist.certification.read/update` | Artist certification |

*Note: Content API and video uploading not supported.*

---

## Implementation: Zod Schemas

```typescript
import { z } from 'zod';

export const IntegrationTypeSchema = z.enum([
  'googlecalendar', 'googledrive', 'gmail', 'googlesheets',
  'googledocs', 'googleslides', 'slack', 'notion',
  'salesforce', 'hubspot', 'linkedin', 'tiktok',
]);

export const ConnectorResourceSchema = z.object({
  type: IntegrationTypeSchema,
  scopes: z.array(z.string()),  // No validation - OAuth provider is the ultimate validator
});

export type ConnectorResource = z.infer<typeof ConnectorResourceSchema>;
```

### Validation Behavior

| Scenario | Behavior |
|----------|----------|
| Unknown integration type | Error: reject file |
| Unknown scope | No validation - OAuth provider validates during consent |
| Empty scopes array | Allowed (some integrations like Notion don't need scopes) |
| Missing `type` field | Error: reject file |

---
## Dependencies

- **Apper**: New `/sync` endpoint - see base44-dev/apper#3374

---

## References

- Apper source: [`backend/app/external_auth/`](https://github.com/base44-dev/apper/tree/main/backend/app/external_auth)
- API endpoints: [`api.py`](https://github.com/base44-dev/apper/blob/main/backend/app/external_auth/api/api.py)
- Data model: [`app_integration.py`](https://github.com/base44-dev/apper/blob/main/backend/app/external_auth/services/app_integration.py)
- Scope enhancement: [`integrations/`](https://github.com/base44-dev/apper/tree/main/backend/app/external_auth/integrations)

### PR DESCRIPTION
**Pull request by Paveltarno**
> [!NOTE]
> ## Description
>
> This PR adds comprehensive support for OAuth connector management in the Base44 CLI. Users can now define OAuth integrations (Google Calendar, Slack, Notion, etc.) in local connector files and sync them to Base44 using the new `base44 connectors push` command, which handles OAuth authorization flows automatically with browser-based authentication and real-time status polling.
>
> ## Related Issue
>
> Closes #184
>
> ## Type of Change
>
> - [ ] Bug fix (non-breaking change which fixes an issue)
> - [x] New feature (non-breaking change which adds functionality)
> - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
> - [ ] Documentation update
> - [ ] Refactoring (no functional changes)
> - [ ] Other (please describe):
>
> ## Changes Made
>
> **Core Infrastructure:**
> - Added connector resource module (`src/core/resources/connector/`) following the existing Resource pattern
> - Implemented Zod schemas for 12 known OAuth providers (Google Calendar, Drive, Gmail, Sheets, Docs, Slides, Slack, Notion, Salesforce, HubSpot, LinkedIn, TikTok)
> - Added support for arbitrary custom OAuth providers with regex validation
> - Implemented file parsing from `connectors/*.json(c)` with duplicate detection
> - Added API client for connector operations (list, set, remove, OAuth status)
>
> **CLI Command:**
> - Implemented `base44 connectors push` command with interactive OAuth flow
> - Automatic browser opening for OAuth authorization
> - Real-time status polling with 2-minute timeout and 2-second intervals
> - Smart CI detection (skips OAuth prompts in CI environments)
> - Comprehensive summary output showing synced/added/removed/failed connectors
>
> **Integration:**
> - Integrated connectors into project config loading pipeline
> - Updated `ProjectData` interface with typed `connectors` field
> - Added `connectorsDir` configuration option (defaults to `connectors/`)
> - Updated AGENTS.md with connector module architecture documentation
>
> **Testing:**
> - Added 360+ lines of unit tests for connector module (`tests/core/connectors.spec.ts`)
> - Added CLI integration tests for push command (`tests/cli/connectors_push.spec.ts`)
> - Added test fixtures for various scenarios (valid connectors, duplicates, invalid files)
> - Added API mock utilities to testkit for connector endpoints
> - All tests pass with comprehensive coverage of sync logic, OAuth flows, and error handling
>
> **Code Quality:**
> - Applied camelCase transformations using Zod `.transform()` for API responses
> - Enhanced error handling with per-connector error tracking
> - Proper handling of edge cases (different user, expired tokens, network failures)
> - Path traversal protection for custom connector types
>
> ## Testing
>
> - [x] I have tested these changes locally
> - [x] I have added/updated tests as needed
> - [x] All tests pass (`npm test`)
>
> ## Checklist
>
> - [x] My code follows the project's style guidelines
> - [x] I have performed a self-review of my own code
> - [x] I have commented my code, particularly in hard-to-understand areas
> - [x] I have made corresponding changes to the documentation (if applicable)
> - [x] My changes generate no new warnings
> - [x] I have updated AGENTS.md if I made architectural changes
>
> ## Additional Notes
>
> **OAuth Flow Details:**
> - The authorization flow automatically opens the user's browser when a connector requires OAuth
> - Polls the OAuth status endpoint every 2 seconds with a 2-minute timeout
> - Handles multiple authorization states: ACTIVE (success), PENDING (timeout), FAILED (error)
> - In CI environments (`process.env.CI`), OAuth flows are automatically skipped
>
> **Sync Behavior:**
> - Syncs local connectors to Base44 (updates scopes if already authorized)
> - Removes remote connectors not present in local project (maintains single source of truth)
> - Handles edge cases: different user already connected, expired tokens, network failures
> - Provides detailed per-connector error messages in the summary
>
> **Architecture:**
> - Follows existing Resource pattern with `readAll` and `push` interface
> - Supports both JSON and JSONC file formats in `connectors/` directory
> - Known providers use predefined schemas; custom providers validated with regex
> - 31 files changed representing a complete feature implementation with tests
>
> ---
> <sub>🤖 Generated by Claude | 2026-02-12 19:21 UTC</sub>

### CODE REVIEW

File: src/core/resources/connector/schema.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  "googlesheets",
+  "googledocs",
+  "googleslides",
+  "slack",
+  "notion",
+  "salesforce",
+  "hubspot",
+  "linkedin",
+  "tiktok",
+] as const;
```
Comment (kfirstri) [inline src/core/resources/connector/schema.ts:115]: Might be annoying that we have to create a new connectorSchema, update the `ConnectorResourceSchema` and add the name to `KnownIntegrationTypes`, Maybe we can enforce this somehow with typescript.. right now the `KnownIntegrationTypes` is not enforced, meaning there can be typos / missing stuff from this list which will cause issues

File: src/core/resources/connector/schema.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+>;
+
+export const SetConnectorResponseSchema = z.object({
+  redirect_url: z.string().nullable(),
+  connection_id: z.string().nullable(),
+  already_authorized: z.boolean(),
+  error: z.string().nullable().optional(),
+  error_message: z.string().nullable().optional(),
+  other_user_email: z.string().nullable().optional(),
+});
```
Comment (kfirstri) [inline src/core/resources/connector/schema.ts]: Use .tranform to transform the response the camelCase (there's examples for this in other resources)

File: src/core/resources/connector/schema.ts [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+export type OAuthStatusResponse = z.infer<typeof OAuthStatusResponseSchema>;
+
+export const RemoveConnectorResponseSchema = z.object({
+  status: z.literal("removed"),
+  integration_type: IntegrationTypeSchema,
+});
+
+export type RemoveConnectorResponse = z.infer<
+  typeof RemoveConnectorResponseSchema
+>;
```
Comment (kfirstri) [inline src/core/resources/connector/schema.ts:196]: related to the entire file: I know we have this issue in many places - can we not export stuff that are not used anywhere? Go over this file and just check what is actually used and export only that

File: src/core/resources/connector/oauth.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+export async function runOAuthFlow(
+  params: OAuthFlowParams
+): Promise<OAuthFlowResult> {
+  await open(params.redirectUrl);
+
+  let finalStatus: ConnectorOAuthStatus = "PENDING";
+
+  await pWaitFor(
```
Comment (kfirstri) [inline src/core/resources/connector/oauth.ts]: polling should not happen in the core/ folder, Core/ should only expose api / schema/ config stuff.
Polling and using `pWaitFor` should happen on the CLI side - you can see an example in `login-flow.ts`

File: src/core/resources/connector/api.ts [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    throw new SchemaValidationError(
+      "Invalid response from server",
+      result.error
+    );
+  }
+
+  return result.data;
+}
+
+export async function setConnector(
```
Comment (kfirstri) [inline src/core/resources/connector/api.ts:40]: nit: maybe we can give this a better name? enableConnector? updateConnector? i feel set is not clear to me

File: src/core/resources/connector/push.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  const upstream = await listConnectors();
+  const localTypes = new Set(connectors.map((c) => c.type));
+
+  for (const connector of connectors) {
+    try {
+      const response = await setConnector(
+        connector.type,
+        connector.scopes ?? []
+      );
+      results.push(setResponseToResult(connector.type, response));
```
Comment (kfirstri) [inline src/core/resources/connector/push.ts]: Maybe it's because i don't like "setConnector" name, but this name is also not clear to me - maybe something like "getConnectorSyncResult"?

File: src/core/resources/connector/push.ts [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+export async function pushConnectors(
+  connectors: ConnectorResource[]
+): Promise<PushConnectorsResponse> {
+  const results: ConnectorSyncResult[] = [];
+  const upstream = await listConnectors();
+  const localTypes = new Set(connectors.map((c) => c.type));
+
+  for (const connector of connectors) {
```
Comment (kfirstri) [inline src/core/resources/connector/push.ts:28]: I think basic flow comments / splitting into function might help in this function. 

// 1. Update local connector to remote
// 2. Delete remote connector that are not in local project

File: src/core/resources/connector/config.ts [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+          hints: [
+            {
+              message: `Remove duplicate connectors with type "${connector.type}" - only one connector per type is allowed`,
+            },
+          ],
+        }
+      );
+    }
+    types.add(connector.type);
+  }
```
Comment (kfirstri) [inline src/core/resources/connector/config.ts:62]: extract to another function called something like `assertNoDuplicateConnectors`

File: src/cli/commands/connectors/push.ts [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+  }
+
+  printSummary(results, oauthOutcomes);
+  return { outroMessage };
+}
+
+export function getConnectorsPushCommand(context: CLIContext): Command {
+  return new Command("push")
+    .description(
+      "Push local connectors to Base44 (syncs scopes and removes unlisted)"
```
Comment (kfirstri) [inline src/cli/commands/connectors/push.ts]: Not sure this description is really clear, "removed unlisted"?

File: src/cli/commands/connectors/push.ts [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+    async () => {
+      return await pushConnectors(connectors);
+    }
+  );
+
+  const oauthOutcomes = new Map<IntegrationType, ConnectorOAuthStatus>();
+  const needsOAuth = results.filter(isPendingOAuth);
+  let outroMessage = "Connectors pushed to Base44";
+
+  if (needsOAuth.length > 0) {
```
Comment (kfirstri) [inline src/cli/commands/connectors/push.ts]: can we flip and exit early?
```
if (needsOauth.length === 0) {
    return { outroMessage };
}
```

File: src/cli/commands/connectors/push.ts [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+    log.info("");
```
Comment (kfirstri) [inline src/cli/commands/connectors/push.ts]: we really need an extra space?

File: src/cli/commands/connectors/push.ts [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+      });
+
+      if (isCancel(shouldAuth) || !shouldAuth) {
+        outroMessage = `Authorization skipped. Pending: ${pending}. Run 'base44 connectors push' again to complete.`;
+      } else {
+        for (const connector of needsOAuth) {
+          log.info(`\nOpening browser for ${connector.type}...`);
+
+          const oauthResult = await runTask(
+            `Waiting for ${connector.type} authorization...`,
```
Comment (kfirstri) [inline src/cli/commands/connectors/push.ts]: Just to add some ticks on the name of the connector
```suggestion
            `Waiting for '${connector.type}' authorization...`,
```

Comment (kfirstri) [reply src/cli/commands/connectors/push.ts]: same for the earlier log

File: src/cli/commands/connectors/push.ts [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+            async () => {
+              return await runOAuthFlow({
```
Comment (kfirstri) [inline src/cli/commands/connectors/push.ts]: I wrote it inside this function but i will mention it here too - the pwaitFor and polling logic should be implemented here, and just call the correct API from the core folder.

File: src/cli/commands/connectors/push.ts [hunk 14]
```diff
@@ -0,0 +1,156 @@
...
+                type: connector.type,
+                redirectUrl: connector.redirectUrl,
+                connectionId: connector.connectionId,
+              });
+            },
+            {
+              successMessage: `${connector.type} authorization complete`,
+              errorMessage: `${connector.type} authorization failed`,
+            }
+          );
```
Comment (kfirstri) [inline src/cli/commands/connectors/push.ts]: consider adding try/catch - if one fails, do we want to entire process to fail or just keep going? i think keep going makes sense..

File: src/core/resources/connector/schema.ts (hunk 1 above)
Comment (Paveltarno) [reply src/core/resources/connector/schema.ts:115]: yeah I agree, talked with Neta about this, we should start syncing the server schema into the CLI but this will happen at a later stage, for now this is mainly a hint for the agent

File: src/core/resources/connector/schema.ts (hunk 3 above)
Comment (Paveltarno) [reply src/core/resources/connector/schema.ts:196]: agreed, how about we add Knip?

Comment (Paveltarno) [reply src/core/resources/connector/schema.ts:196]: added

File: src/cli/commands/connectors/push.ts (hunk 11 above)
Comment (Paveltarno) [reply src/cli/commands/connectors/push.ts]: Fixed in 65d3b25a1221e690b3d037fb341d0fe89c9ff7b3

File: src/cli/commands/connectors/push.ts (hunk 10 above)
Comment (Paveltarno) [reply src/cli/commands/connectors/push.ts]: Fixed in 65d3b25a1221e690b3d037fb341d0fe89c9ff7b3

File: src/cli/commands/connectors/push.ts (hunk 12 above)
Comment (Paveltarno) [reply src/cli/commands/connectors/push.ts]: Fixed in 65d3b25a1221e690b3d037fb341d0fe89c9ff7b3

File: src/core/resources/connector/push.ts (hunk 7 above)
Comment (Paveltarno) [reply src/core/resources/connector/push.ts:28]: Fixed in 65d3b25a1221e690b3d037fb341d0fe89c9ff7b3

File: src/core/resources/connector/schema.ts (hunk 2 above)
Comment (Paveltarno) [reply src/core/resources/connector/schema.ts]: Fixed in 65d3b25a1221e690b3d037fb341d0fe89c9ff7b3

File: src/cli/commands/connectors/push.ts (hunk 9 above)
Comment (Paveltarno) [reply src/cli/commands/connectors/push.ts]: Fixed in 65d3b25a1221e690b3d037fb341d0fe89c9ff7b3

File: src/core/resources/connector/push.ts (hunk 6 above)
Comment (Paveltarno) [reply src/core/resources/connector/push.ts]: Fixed in 65d3b25a1221e690b3d037fb341d0fe89c9ff7b3 - renamed to getConnectorSyncResult

File: src/core/resources/connector/config.ts (hunk 8 above)
Comment (Paveltarno) [reply src/core/resources/connector/config.ts:62]: Fixed in 65d3b25a1221e690b3d037fb341d0fe89c9ff7b3

File: src/core/resources/connector/api.ts (hunk 5 above)
Comment (Paveltarno) [reply src/core/resources/connector/api.ts:40]: for now keps as setConnector - it mirrors the API endpoint name (PUT /integrations/{type})

File: src/cli/commands/connectors/push.ts (hunk 14 above)
Comment (Paveltarno) [reply src/cli/commands/connectors/push.ts]: Fixed in 65d3b25a1221e690b3d037fb341d0fe89c9ff7b3

File: src/core/resources/connector/schema.ts [hunk 15]
```diff
@@ -0,0 +1,196 @@
+import { z } from "zod";
```
Comment (kfirstri) [inline src/core/resources/connector/schema.ts:1]: Can we go over all the core/ tests and clear some stuff? i think testing the schema is pointless
